### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by AnHeuermann**
### Related Issues

Building OpenModelica from source is complicated and by times frustrating.
The Readmes are outdated and hard to follow.
See e.g. https://github.com/OpenModelica/OpenModelica/issues/10151 for a ticket where a user struggled with the README for WSL.

### Purpose

Update READMEs to make build process more clear.

### Changes

- Added CMake build to compilation instructions
- Added table of content, unified layout of Windows and Linux instructions
- Added OSMC and citation section to main README
- Removed WSL README. It's a Linux OS after all
- Removed unused (?) Windows build instructions

### CODE REVIEW

File: OMCompiler/README.Linux.md [hunk 1]
```diff
@@ -58,7 +85,57 @@ But first you need to install dependencies:
 - [Sundials](http://www.llnl.gov/CASC/sundials/) (optional; adds more numerical solvers to the simulation runtime)
 - libcurl (libcurl4-gnutls-dev)
 
-# Setting your environment for compiling OpenModelica
+# 2 Compile OpenModelica
+
+Now there are two options to build OpenModelica:
+
+  1. Use new CMake build.
```
Comment (casella) [inline OMCompiler/README.Linux.md:92]: It's probably not clear to people outside the developers' inner circle what are the pro's and con's of each choice. Maybe you can quickly summarize them.

**Review by casella — APPROVED** [review]
Very nice job. I just added a suggestion about CMake vs. Makefile, you may want to add something there.

Of course the proof is in the pudding, we should have some volunteers trying out the instructions themselves to see if they work. I guess that will come by itself in due time 😄

File: OMCompiler/README.Linux.md (hunk 1 above)
Comment (AnHeuermann) [reply OMCompiler/README.Linux.md:92]: Newer is always better? The first is always the best?
But yes, I'll recommend CMake.

Comment (casella) [reply OMCompiler/README.Linux.md:92]: Seriously, is there any reason why we still keep the old makefile? Is it just in case CMake has issues? Or there are some cases where it is still preferable. Maybe @mahge can comment on that.

**Comment by AnHeuermann** [comment]
Thanks @anotheruserofgithub. Most errors should be fixed now. Types are probably still there 😆

File: OMCompiler/README.Linux.md (hunk 1 above)
Comment (mahge) [reply OMCompiler/README.Linux.md:95]: There are some users/developers who prefer the `Makefiles` build system. Some of our CI/CD scripts such as those generating packages for Linux and Windows still uses the `Makefiles` build system. It is not that it is preferable (I guess we now have `CMake` everywhere we need it) it is just that the scripts need to updated and tested properly if we are to switch them.

My recommendation is to make sure that **we point new users to the `CMake` build system**. @AnHeuermann is doing that with these updates. 

We can also start replacing the `Makefiles` based CI/CD scripts we have one by one. It should not be a lot of work.

I even tried CMake based packaging for Windows and Debian Linux a while back and everything seems to work fine. So we can also update the packaging processes. Most of the required additions/modifications are already done when we did the packaging support for macOS. So it should be a much easier transition now.

Comment (perost) [reply OMCompiler/README.Linux.md:95]: I think there's still an issue that OMEdit built with CMake is slow, at least for me. It takes like 20 sec to start, most of which is loading the MSL. I'm not really sure what the issue is, some bad optimization flags? But that's probably something we need to fix before starting to use CMake for packaging.

Comment (mahge) [reply OMCompiler/README.Linux.md:95]: Oh right. There is that. I am not really sure what the issue is there. Using `omc `on the `commandline` and loading Modelica is fast enough. The `omc` executable and the `OMEdit` executable both use the same exact shared library `libOpenModelicaCompiler.so` to load Modelica libraries. So perhaps the issue lies somewhere in how the OMEdit source code is compiled. There is probably some flag that is manually added to the Makefiles with such significant importance.  

The delay happens only on Linux. On Windows, IIRC, there is no such issue. I should probably investigate this properly since I got used to it and do not even notice it anymore. Which is not good.

**Comment by adrpo** [comment]
As for the NSIS part, we can just remove it or we could just point the users to this below.

We still use the makefiles and NSIS to build the releases.  We can point the users to:
https://github.com/OpenModelica/OpenModelicaSetup/blob/master/BuildWindowsRelease.sh
and the batch that we use to call it from:
https://github.com/OpenModelica/OpenModelicaSetup/tree/master/batch

File: OMCompiler/README.Linux.md (hunk 1 above)
Comment (casella) [reply OMCompiler/README.Linux.md:95]: OK. I'm fine at directing new users to CMake. My point is that we should include a (brief) honest assessment of the pro's and con's of the two systems, otherwise most people who are not deep into the thing will have no clue.
